composer/utils/ddp.py

```diff
@@ -1,15 +1,52 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
+import os
 import warnings
+from typing import Callable, Optional
 
 import torch.distributed as dist
 
 
-def get_global_rank() -> int:
+def _get_distributed_config_var(env_var: str,
+                                human_name: str,
+                                default: int,
+                                fetch_fn: Optional[Callable[[], int]] = None) -> int:
     if not dist.is_available():
-        warnings.warn("Torch distributed is not available; returning 0 for global rank")
-        return 0
-    return dist.get_rank()  # raises a runtime error if not yet initialized
+        warnings.warn(f"Torch distributed is not available; returning {default} for {human_name}")
+        return default
+
+    if fetch_fn is not None:
```

Comment:
In an environment where torch.distributed is not available, then I believe this would throw an attributeerror before `_get_distributed_config_var` is called. Instead, perhaps provide the fetch_fn by string and do a `getattr(torch.dist, fetch_fn_name)()`? This should not raise an attributeerror here cause of the check on line 14.